src/Nethermind/Nethermind.Core.Test/Blockchain/TestBlockchain.cs

```diff
@@ -172,28 +172,39 @@ protected virtual async Task<TestBlockchain> Build(ISpecProvider specProvider =
 
             var genesis = GetGenesisBlock();
             BlockTree.SuggestBlock(genesis);
-            await _resetEvent.WaitAsync();
-            //if (!await _resetEvent.WaitAsync(1000))
-            // {
-            //     throw new InvalidOperationException("Failed to process genesis in 1s.");
-            // }
-
+            await WaitAsync(_resetEvent, "Failed to process genesis in time.");
             await AddBlocksOnStart();
             return this;
         }
 
+        private async Task WaitAsync(SemaphoreSlim semaphore, string error, int timeout = 1000)
```

Comment:
we have a helper class for it right? AsyncWait or sth

Reply:
its used here in 2nd overload, we have it for WaitHandle not for Semaphore